src/control/positionControl/pathPlanning/VoronoiPathPlanning.cpp

```diff
@@ -0,0 +1,153 @@
+//
+// Created by ratoone on 05-11-19.
+//
+
+#include "control/positionControl/pathPlanning/VoronoiPathPlanning.h"
+
+namespace rtt::ai::control{
+
+VoronoiPathPlanning::VoronoiPathPlanning(CollisionDetector& collisionDetector) : collisionDetector(collisionDetector){}
+
+void VoronoiPathPlanning::computeDiagram(const rtt::Vector2 &robotPosition, const rtt::Vector2 &targetPosition) {
+    auto robots = collisionDetector.getRobotPositions();
+    std::vector<jcv_point> robotPositions(robots.size());
+    std::transform(robots.begin(), robots.end(), robotPositions.begin(),
+            [](auto robot)-> jcv_point {return {(float)robot->x, (float)robot->y};});
+    jcv_rect playArea = {
+            (jcv_point){(float)std::min(robotPosition.x,targetPosition.x),(float)std::min(robotPosition.y,targetPosition.y)},
+            (jcv_point){(float)std::max(robotPosition.x,targetPosition.x),(float)std::max(robotPosition.y,targetPosition.y)}
```

Comment:
I can tell you from experience that many optimal paths are not limited to the rectangle defined by the targetPosition and the robotPosition. This is certainly not sufficient for object avoidance.

Reply:
Indeed, but in the end we may not even use Voronoi. If someone decides this is needed, it can be optimized then.